tests/treactiveSigil.nim

```diff
@@ -0,0 +1,38 @@
+import sigils/reactive
+
+import unittest
+import std/sequtils
+
+suite "reactive examples":
+
+  test "reactive wrapper":
+
+    let
+      x = reactive(5)
+      y = computed[int]():
+        2 * x{}
+
+    check x.val == 5
+    check y.val == 0
+    x <- 2
+    check y.val == 4
+    # x <- 2
+
+  test "reactive wrapper trace executions and side effects":
+
+    var cnt = Sigil[int](val: 0)
+
+    let
+      x = reactive(5)
+      z = computed[int]():
+        cnt.val.inc()
+        8 * x{}
+
+    check cnt.val == 1 # cnt is called from the `read` (`trace`) setup step
```

Comment:
Out of curiosity, would a computed derived from multiple differently typed signals work?
Just wrote myself a test-case out of curiosity but that seems to fail for `z` after I set `y` to true
```nim
  test "reactive wrapper multiple signals":
    let x = reactive(5)
    let y = reactive(false)
    let z = computed[int]():
      if y{}:
        x{} * 2
      else:
        0
    
    check x.value == 5
    check y.value == false
    check z.value == 0
    
    y <- true
    check y.value == true
    check z.value == 5
    
    x <- 2
    check x.value == 2
    check z.value == 4
    
    y <- false
    check y.value == false
    check z.value == 0
```

Reply:
Hmmmm, another good case. The value is actually gotten from the "deref" `x{}` right? 

Actually that probable means I'd want to change the signal from `proc changed*[T](r: Sigil[T], val: T) {.signal.}` to `proc changed*[T](r: Sigil[T]) {.signal.}`.

Reply:
That actually makes sense to me. Because the sigil itself contains the value that changed, therefore it is not necessary to also transport the value in a separate param (from my naive point of view, I'm still not 100% on grasping how everything interacts)